src/sagemaker/serializers.py

```diff
@@ -357,3 +356,38 @@ def serialize(self, data):
             return data.read()
 
         raise ValueError("Unable to handle input format: %s" % type(data))
+
+
+class DataSerializer(SimpleBaseSerializer):
+    """Serialize data in any file by extracting raw bytes from the file."""
+
+    def __init__(self, content_type="file-path/raw-bytes"):
+        """Initialize a ``DataSerializer`` instance.
+
+        Args:
+            content_type (str): The MIME type to signal to the inference endpoint when sending
+                request data (default: "file-path/raw-bytes").
+        """
+        super(DataSerializer, self).__init__(content_type=content_type)
+
+    def serialize(self, data):
+        """Serialize file data to a raw bytes.
+
+        Args:
+            data (object): Data to be serialized. The data can be a string,
+                representing file-path or the raw bytes from a file.
+        Returns:
+            raw-bytes: The data serialized as a raw-bytes from the input.
+        """
+        if isinstance(data, str):
+            try:
+                dataFile = open(data, "rb")
+            except Exception:
+                raise ValueError(f"{data} is not a valid file-path.")
```

Comment:
Lets generalise the exception and not conclude `not a valid file-path` for better debugging.
```
try:
	dataFile = open(data, "rb")
	dataFileInfo = dataFile.read()
	dataFile.close()
except Exception as e:
         raise ValueError(f"Could not open/read file: {data}. {e.message}")
```

Reply:
updated in the current revision.